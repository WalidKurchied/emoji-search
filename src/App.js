import React, { Component } from 'react';
import Header from './components/Header';
import SearchInput from './components/SearchInput';
import EmojiResults from './components/EmojiResults';
import emojis from './emojiList.json';

class App extends Component {
  state = {
    emojiList: []
  };

  componentDidMount() {
    this.filterEmojis();
  }

  filterEmojis = (searchKeyword = null) => {
    let maxEmojis = 1;

    const results = emojis.filter(emoji => {        
      if ((!searchKeyword || emoji.keywords.includes(searchKeyword)) && maxEmojis <= 20) {
        maxEmojis++;
        return true;
      }  

      return false;
    });

    this.setState({emojiList: results});
  }

  render() {
    return (
      <div className="App">
        <Header />
        <SearchInput filterEmojis={this.filterEmojis} />
        <EmojiResults emojiList={this.state.emojiList} />
      </div>
    );
  }
}

export default App;