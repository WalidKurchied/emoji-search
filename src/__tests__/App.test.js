import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

describe('<App />', () => {
  let component = null;

  beforeEach(() => {
    component = create(<App />);
  });

  it('Should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should display a list of emojis', () => {
    const instance = component.getInstance();
    expect(instance.state.emojiList.length).toBe(20);
  });

  it('Should change emojilist state value based on the filterEmojis search query', () => {
    const instance = component.getInstance();
    expect(instance.state.emojiList.length).toBe(20);

    instance.filterEmojis('earth');

    expect(instance.state.emojiList.length).toBe(3);
    expect(instance.state.emojiList[0].title).toBe('Earth Americas');
  });
});
