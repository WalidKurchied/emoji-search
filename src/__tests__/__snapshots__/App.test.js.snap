// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match snapshot 1`] = `
<div
  className="App"
>
  <header
    id="header"
  >
    <img
      alt="header emoji one"
      className="header-emoji"
      height="32"
      src="//cdn.jsdelivr.net/emojione/assets/png/1f638.png"
      width="32"
    />
    <h1
      id="emojiAppTitle"
    >
      Emoji Search
    </h1>
    <img
      alt="header emoji two"
      className="header-emoji"
      height="32"
      src="//cdn.jsdelivr.net/emojione/assets/png/1f63a.png"
      width="32"
    />
  </header>
  <div
    id="searchInputContainer"
  >
    <input
      id="searchInput"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    id="emojiResults"
  >
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="100"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f4af.png"
      />
      <span
        className="emoji-title"
      >
        100
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="1234"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f522.png"
      />
      <span
        className="emoji-title"
      >
        1234
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Grinning"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f600.png"
      />
      <span
        className="emoji-title"
      >
        Grinning
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Grimacing"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f62c.png"
      />
      <span
        className="emoji-title"
      >
        Grimacing
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Grin"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f601.png"
      />
      <span
        className="emoji-title"
      >
        Grin
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Joy"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f602.png"
      />
      <span
        className="emoji-title"
      >
        Joy
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Smiley"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f603.png"
      />
      <span
        className="emoji-title"
      >
        Smiley
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Smile"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f604.png"
      />
      <span
        className="emoji-title"
      >
        Smile
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Sweat Smile"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f605.png"
      />
      <span
        className="emoji-title"
      >
        Sweat Smile
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Laughing"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f606.png"
      />
      <span
        className="emoji-title"
      >
        Laughing
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Innocent"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f607.png"
      />
      <span
        className="emoji-title"
      >
        Innocent
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Wink"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f609.png"
      />
      <span
        className="emoji-title"
      >
        Wink
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Blush"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f60a.png"
      />
      <span
        className="emoji-title"
      >
        Blush
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Slight Smile"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f642.png"
      />
      <span
        className="emoji-title"
      >
        Slight Smile
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Upside Down"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f643.png"
      />
      <span
        className="emoji-title"
      >
        Upside Down
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Relaxed"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/263a.png"
      />
      <span
        className="emoji-title"
      >
        Relaxed
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Yum"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f60b.png"
      />
      <span
        className="emoji-title"
      >
        Yum
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Relieved"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f60c.png"
      />
      <span
        className="emoji-title"
      >
        Relieved
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Heart Eyes"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f60d.png"
      />
      <span
        className="emoji-title"
      >
        Heart Eyes
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
    <div
      className="emoji-row"
      onClick={[Function]}
    >
      <img
        alt="Kissing Heart"
        className="emoji-icon"
        src="//cdn.jsdelivr.net/emojione/assets/png/1f618.png"
      />
      <span
        className="emoji-title"
      >
        Kissing Heart
      </span>
      <span
        className="copy-info"
      >
        Click to copy emoji
      </span>
    </div>
  </div>
</div>
`;
